src/main/java/net/schmizz/sshj/transport/NegotiatedAlgorithms.java

```diff
@@ -15,37 +15,38 @@
  */
 package net.schmizz.sshj.transport;
 
-import java.util.List;
-
 public final class NegotiatedAlgorithms {
 
     private final String kex;
-    private final List<String> availableSigs;
+    private final String sig;
     private final String c2sCipher;
     private final String s2cCipher;
     private final String c2sMAC;
     private final String s2cMAC;
     private final String c2sComp;
     private final String s2cComp;
 
-    NegotiatedAlgorithms(String kex, List<String> availableSigs, String c2sCipher, String s2cCipher, String c2sMAC, String s2cMAC,
-                         String c2sComp, String s2cComp) {
+    private final Boolean rsaSHA2Support;
```

Comment:
Why is this not a primitive `boolean`?

Reply:
No reason, I'm just not writing much Java these days. Fixed.